src/main/java/ViewModel/musicEngineControllerViewModel.java

```diff
@@ -31,6 +32,21 @@ public void getRecommendationAction(int id){
         this.musicEngineController.playRecommendation(data);
     }
 
+    public void pauseSongAction(){
+        this.musicEngineController.pauseSong();
+    }
+
+    public void skipSongAction(){
+        this.musicEngineController.playNext();
+    }
+
+    public String playSongAction(int id){
+        songInputData data = new songInputData(id);
```

Comment:
IIRC we were creating the InputData in the Controller. Technically, this does not violate clean architecture but I would prefer that we stay consistent. This may be another thing we should discuss (along with where we are constructing the interactor -- the ViewModel or the Controller) and we might want to open an Issue for refactoring this later once all of our use cases are functional.

Reply:
fixed